File: 346-moving-average-from-data-stream/moving-average-from-data-stream.java
class MovingAverage {
    LinkedList<Integer> list;
    int sum = 0;
    int window ;
    public MovingAverage(int size) {
        this.window = size;
        list = new LinkedList<>();
    }
    
    public double next(int val) {
        if (list.size() < window) {
            sum += val;
            list.add(val);
            return sum * 1.0 / list.size();
        } else {
            sum -= list.removeFirst();
            sum += val;
            list.add(val);
            return sum * 1.0 / window;
        }
    }
}

/**
 * Your MovingAverage object will be instantiated and called as such:
 * MovingAverage obj = new MovingAverage(size);
 * double param_1 = obj.next(val);
 */

File: 346-moving-average-from-data-stream/moving-average-from-data-stream.py
class MovingAverage:

    def __init__(self, size: int):
        self.sz = size
        self.queue = deque()
        self.wsum = 0

    def next(self, val: int) -> float:
        self.queue.append(val)
        self.wsum += val
        if len(self.queue) > self.sz:
            self.wsum -= self.queue.popleft()
        return self.wsum / len(self.queue)
        


# Your MovingAverage object will be instantiated and called as such:
# obj = MovingAverage(size)
# param_1 = obj.next(val)